lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -1370,13 +1395,16 @@ CreateRunInTerminalReverseRequest(const llvm::json::Object &launch_request,
   if (!cwd.empty())
     run_in_terminal_args.try_emplace("cwd", cwd);
 
-  // We need to convert the input list of environments variables into a
-  // dictionary
-  std::vector<std::string> envs = GetStrings(launch_request_arguments, "env");
+  std::unordered_map<std::string, std::string> envMap =
+      GetStringMap(*launch_request_arguments, "env");
```

Comment:
You are completely dropping for support for an array-based env list, and we can't do this because we'd be breaking tons of launch.json configs. You need to effectively support both formats: a map and an array. Convert `GetStringMap` into `GetEnv` that takes into account both formats.